packages/web-api-scan-runner/src/crawler/discovered-urls-processor.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import _, { isNil } from 'lodash';
+import { WebsiteScanResult } from 'storage-documents';
+
+@injectable()
+export class DiscoveredUrlsProcessor {
+    public constructor(@inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration) {}
+
+    public async getProcessedUrls(discoveredUrls: string[], websiteScanResult?: WebsiteScanResult): Promise<string[]> {
+        let processedUrls = discoveredUrls;
+        if (isNil(discoveredUrls)) {
+            return [];
+        }
+
+        if (!isNil(websiteScanResult) && !isNil(websiteScanResult.knownPages)) {
```

Comment:
Would optional chaining work here? (i.e., the ?. operator) It would take this fown from one condition to two.